### ISSUE TITLE
Use `NonZeroUsize` rather than `usize` to prevent panic scenario

### ISSUE THREAD

**Issue by mx00s** [schell/mogwai#97]
A [`NonZeroUsize`](https://doc.rust-lang.org/std/num/struct.NonZeroUsize.html) parameter instead of a `usize`to specify the capacity of a bounded channel would eliminate the [panic scenario](https://docs.rs/mogwai/0.5.3/mogwai/channel/broadcast/fn.bounded.html#panics).

**Comment by schell** [comment]
Hey! Great idea.

**Comment by schell** [comment]
It might take me a bit to get to this, I'm currently on holiday. If you're feeling antsy I would definitely accept a pull request.

**Comment by schell** [comment]
I tried this out and although it is more correct to use a NonZeroUsize for the capacity, it is quite inconvenient. I also found that I just ended up calling `1.try_into().unwrap()` while creating all of my channels and while this is more explicit (which I like) it was a pain. I temporarily included a constant `const ONE: NonZeroUsize = ...` but ultimately decided not to include this change in `0.6`. 

I might pick it back up for `0.7`, but I might also just remove the channels in favor of `Input`, `FanInput` and `Output`.
